### ISSUE TITLE
The filter hook woocommerce_loop_add_to_cart_link doesn't work

### ISSUE THREAD

**Issue by lomars** [woocommerce/woocommerce#42671]
The filter hook `woocommerce_loop_add_to_cart_link` doesn't work on WooCommerce Blocks involving a product loop like Best selling products,  Products by category …

For example I am using this hooked function:
```
add_filter( 'woocommerce_loop_add_to_cart_link', 'replace_loop_add_to_cart_button', 10, 2 );
function replace_loop_add_to_cart_button( $button, $product  ) {
    return '<a class="button" href="' . $product->get_permalink() . '">' . __( "View product", "woocommerce" ) . '</a>';
}
```
It works everywhere on WooCommerce product loops, on WooCommerce shortcode `[products]` replacing add to cart button with a button linked to the single product page. But has no effect in WooCommerce Blocks involving products loop.

Using latest WordPress, WooCommerce and WooCommerce Blocks versions.

**Comment by nerrad** [comment]
This is something that will be considered as a part of the work with the product query block and revamping of the existing product archive type blocks.

**Comment by tnchuntic** [comment]
Any updates on this one? It seems that plugin "https://woocommerce.com/products/request-a-quote-plugin-for-woocommerce/" is affected by this issue also.

**Comment by solaceten** [comment]
Just came across this issue. 

Well, two 2.5 years later, since the original post and 1 year since @nerrad's comment.... no resolution, that's a tad disappointing. 

I am using Featured Products and I can't figure out how to change the behaviour of "Add To Cart" -> "View Product"
Is there an alternate method I've missed?

**Comment by soulstyle** [comment]
Any updates on this? I'm also trying to do similar things as @solaceten

**Comment by Aljullu** [comment]
This seems to be working fine with the Product Collection block, can any of you folks check? I just tested the snippet from @lomars and it's behaving as expected:

![Image](https://github.com/user-attachments/assets/6d9d0f0d-1280-408c-94fc-1fa8f97888ef)

As mentioned in [this post](https://developer.woocommerce.com/2024/11/11/product-collection-block-new-js-events-and-soft-deprecation-of-product-grid-blocks/), all other product grid blocks have been soft-deprecated in favor of the Product Collection block.

**Comment by ninetyninew** [comment]
It works because of the compatibility layer.

However if the product collection isn't in a block template and just a standalone block in the homepage, for example, the hooks won't be there to use.

So it needs some extra coding to get it to work in that scenario

**Comment by Aljullu** [comment]
@ninetyninew I tested it on a regular page, not a template. 🤔 

_Editor:_

![Image](https://github.com/user-attachments/assets/776aee4a-0ec0-4f68-b82f-937bfeca5464)

_Frontend:_

![Image](https://github.com/user-attachments/assets/d68ddb75-479d-44f3-ad04-ff58f66e1c8b)

**Comment by ninetyninew** [comment]
Ah sorry, I thought this was a different hook question, as they are using "woocommerce_loop_add_to_cart_link" i think it will be fine.

However, if they were trying to do something off of "woocommerce_after_shop_loop_item" they'd run into the issue with standalone blocks on non-template pages as per my previous comment.

**Comment by Aljullu** [comment]
True, I think it might be better to create a separate issue for that and close this one to avoid confusion. What do you think, @ninetyninew?

**Comment by Aljullu** [comment]
> True, I think it might be better to create a separate issue for that and close this one to avoid confusion. What do you think, @ninetyninew?

A GitHub Discussion has been opened to evaluate enabling the compatibility layer in non-template pages: https://github.com/woocommerce/woocommerce/discussions/52915. I will close this issue in favor of that discussion, hope that ok for everybody!
